Content.Server/GameObjects/EntitySystems/SpawnAfterInteractSystem.cs

```diff
@@ -0,0 +1,66 @@
+using Content.Server.GameObjects.Components.Engineering;
+using Content.Server.GameObjects.Components.Stack;
+using Content.Server.GameObjects.EntitySystems.DoAfter;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Content.Shared.Utility;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class SpawnAfterInteractSystem : EntitySystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SpawnAfterInteractComponent, AfterInteractMessage>(HandleAfterInteract);
+        }
+
+        private async void HandleAfterInteract(EntityUid uid, SpawnAfterInteractComponent component, AfterInteractMessage args)
+        {
+            if (component.Prototype == null)
+                return;
+            if(!_mapManager.TryGetGrid(args.ClickLocation.GetGridId(component.Owner.EntityManager), out var grid))
+                return;
+            if (!args.InRangeUnobstructed(ignoreInsideBlocker: true, popup: true))
+                return;
+            var snapPos = grid.SnapGridCellFor(args.ClickLocation, SnapGridOffset.Center);
+
+            bool CheckTileClear()
+            {
+                return !grid.GetTileRef(snapPos).Tile.IsEmpty && args.User.InRangeUnobstructed(args.ClickLocation);
+            }
```

Comment:
This can SURELY use one of the `TurfHelpers` extensions instead. If not, please write a new one as this code will probably be repeated a lot thorough the codebase...